test/e2e/gardener/shoot/create_hibernate_wakeup_delete.go

```diff
@@ -5,128 +5,149 @@
 				// might increase the test duration undesirably.
-				node.VerifyNodeCriticalComponentsBootstrapping(parentCtx, f.ShootFramework)
+				node.VerifyNodeCriticalComponentsBootstrapping(s)
 			}
 
-			By("Hibernate Shoot")
-			ctx, cancel = context.WithTimeout(parentCtx, 10*time.Minute)
-			defer cancel()
-			Expect(f.HibernateShoot(ctx, f.Shoot)).To(Succeed())
+			It("Hibernate Shoot", func(ctx SpecContext) {
+				Eventually(ctx, func() error {
+					patch := client.MergeFrom(s.Shoot.DeepCopy())
+					s.Shoot.Spec.Hibernation = &gardencorev1beta1.Hibernation{
+						Enabled: ptr.To(true),
+					}
+
+					return s.GardenClient.Patch(ctx, s.Shoot, patch)
+				}).Should(Succeed())
+			}, SpecTimeout(time.Minute))
 
-			By("Wake up Shoot")
-			ctx, cancel = context.WithTimeout(parentCtx, 20*time.Minute)
-			defer cancel()
-			Expect(f.WakeUpShoot(ctx, f.Shoot)).To(Succeed())
+			ItShouldWaitForShootToBeReconciledAndHealthy(s)
 
-			if !v1beta1helper.IsWorkerless(f.Shoot) {
-				inclusterclient.VerifyInClusterAccessToAPIServer(parentCtx, f.ShootFramework)
-			}
+			It("Wake up Shoot", func(ctx SpecContext) {
+				Eventually(ctx, func() error {
+					patch := client.MergeFrom(s.Shoot.DeepCopy())
+					s.Shoot.Spec.Hibernation = &gardencorev1beta1.Hibernation{
+						Enabled: ptr.To(false),
+					}
 
-			By("Delete Shoot")
-			ctx, cancel = context.WithTimeout(parentCtx, 20*time.Minute)
-			defer cancel()
-			Expect(f.DeleteShootAndWaitForDeletion(ctx, f.Shoot)).To(Succeed())
-		})
-	}
+					return s.GardenClient.Patch(ctx, s.Shoot, patch)
+				}).Should(Succeed())
+			}, SpecTimeout(time.Minute))
 
-	Context("Shoot with workers", Label("basic"), func() {
-		test(e2e.DefaultShoot("e2e-wake-up"))
-	})
+			ItShouldWaitForShootToBeReconciledAndHealthy(s)
 
-	Context("Workerless Shoot", Label("workerless"), func() {
-		test(e2e.DefaultWorkerlessShoot("e2e-wake-up"))
-	})
+			// TODO(timebertt): add inclusterclient.VerifyInClusterAccessToAPIServer once it got refactored
 
-	Context("Shoot with workers with NamespacedCloudProfile", Label("basic"), func() {
-		shoot := e2e.DefaultShoot("e2e-wake-up-ncp")
-		shoot.Spec.CloudProfile = &gardencorev1beta1.CloudProfileReference{
-			Kind: "NamespacedCloudProfile",
-			Name: "my-profile",
+			ItShouldDeleteShoot(s)
+			ItShouldWaitForShootToBeDeleted(s)
 		}
-		test(shoot)
+
+		Context("Shoot with workers", Label("basic"), Ordered, func() {
+			test(NewTestContext().ForShoot(DefaultShoot("e2e-wake-up")))
+		})
+
+		Context("Workerless Shoot", Label("workerless"), Ordered, func() {
+			test(NewTestContext().ForShoot(DefaultWorkerlessShoot("e2e-wake-up")))
+		})
+
+		Context("Shoot with workers with NamespacedCloudProfile", Label("basic"), Ordered, func() {
+			shoot := DefaultShoot("e2e-wake-up-ncp")
+			shoot.Spec.CloudProfile = &gardencorev1beta1.CloudProfileReference{
+				Kind: "NamespacedCloudProfile",
+				Name: "my-profile",
+			}
+
+			s := NewTestContext().ForShoot(shoot)
```

Comment:
Can you wrap this in a `BeforeTestSetup` block that initializes a `ShootContext` variable, but doesn't expose the `shoot` variable?